Geometry/MTDGeometryBuilder/src/MTDParametersFromDD.cc

```diff
@@ -46,32 +48,61 @@ bool MTDParametersFromDD::build(const DDCompactView* cvp, PMTDParameters& ptp) {
   DDSpecificsHasNamedValueFilter filter1{attribute};
   DDFilteredView fv1(*cvp, filter1);
   bool ok = fv1.firstChild();
+  int topoMode(-1);
   if (ok) {
     DDsvalues_type sv(fv1.mergedSpecifics());
-    int topoMode = getMTDTopologyMode("TopologyMode", sv);
+    topoMode = getMTDTopologyMode("TopologyMode", sv);
     ptp.topologyMode_ = topoMode;
   } else {
     throw cms::Exception("MTDParametersFromDD") << "Not found " << attribute.c_str() << " but needed.";
   }
 
+  if (topoMode >= static_cast<int>(MTDTopologyMode::Mode::btlv1etlv5)) {
+    std::array<std::string, 8> etlLayout{{
+        "StartCopyNo_Front_Left",
+        "StartCopyNo_Front_Right",
+        "StartCopyNo_Back_Left",
+        "StartCopyNo_Back_Right",
+        "Offset_Front_Left",
+        "Offset_Front_Right",
+        "Offset_Back_Left",
+        "Offset_Back_Right",
+    }};
+    int sector(0);
+    for (const auto& name : etlLayout) {
+      auto const& v = cvp->vector(name);
+      if (!v.empty()) {
+        sector++;
+        std::vector<int> ipos = dbl_to_int(v);
+        putOne(sector, ipos, ptp);
+      } else {
+        throw cms::Exception("MTDParametersFromDD") << "Not found " << name << " but needed.";
+      }
+    }
+  }
+
   return true;
 }
 
 bool MTDParametersFromDD::build(const cms::DDCompactView* cvp, PMTDParameters& ptp) {
   cms::DDVectorsMap vmap = cvp->detector()->vectors();
 
   std::array<std::string, 2> mtdSubdet{{"BTL", "ETL"}};
-  int subdet(0);
+  int subdet(0), oldsubdet(-1);
   for (const auto& name : mtdSubdet) {
-    subdet += 1;
     for (auto const& it : vmap) {
       if (dd4hep::dd::compareEqual(dd4hep::dd::noNamespace(it.first), name)) {
+        subdet++;
+        oldsubdet = subdet;
         std::vector<int> subdetPars;
         for (const auto& i : it.second)
           subdetPars.emplace_back(std::round(i));
         putOne(subdet, subdetPars, ptp);
       }
     }
+    if (oldsubdet != subdet) {
```

Comment:
Wouldn't it be simpler to just check here:
```
if (subdet == 0) {
```
Then `oldsubdet` wouldn't even be needed.

Reply:
well, ```subdet``` could be 1 because just one item is found, but still there is an inconsistency in the input. Anyway the logic is not working in this case, as ```oldsubdet``` should be reset inside the main loop. A ```bool``` may do the same, I will update it.

Reply:
@cvuosalo with the latest update, and the verbosity activated, I get:

```
%MSG-i MTDParametersFromDD:  PreProcPath p1 07-May-2021 11:23:02 CEST  Run: 1 Event: 1
Adding PMTDParameters item: 1 with 12 elements: 0 0 0 0 3733 1867 112 60 1 16 3 1
%MSG
%MSG-i MTDParametersFromDD:  PreProcPath p1 07-May-2021 11:23:02 CEST  Run: 1 Event: 1
Adding PMTDParameters item: 2 with 12 elements: 0 0 0 0 50 50 50 50 16 16 2 1
%MSG
%MSG-i MTDParametersFromDD:  PreProcPath p1 07-May-2021 11:23:02 CEST  Run: 1 Event: 1
Adding PMTDParameters item: 11 with 28 elements: 1 8 21 37 55 75 97 120 144 169 193 213 233 252 271 291 311 332 357 382 406 428 449 468 485 500 511 517
%MSG
%MSG-i MTDParametersFromDD:  PreProcPath p1 07-May-2021 11:23:02 CEST  Run: 1 Event: 1
Adding PMTDParameters item: 12 with 28 elements: 1 7 18 33 50 69 90 112 136 161 186 207 227 247 266 285 305 325 349 374 398 421 443 463 481 497 510 517
%MSG
%MSG-i MTDParametersFromDD:  PreProcPath p1 07-May-2021 11:23:02 CEST  Run: 1 Event: 1
Adding PMTDParameters item: 13 with 28 elements: 1 4 14 28 45 64 85 107 130 154 179 201 221 241 260 280 300 320 343 368 392 415 437 457 475 491 504 513
%MSG
%MSG-i MTDParametersFromDD:  PreProcPath p1 07-May-2021 11:23:02 CEST  Run: 1 Event: 1
Adding PMTDParameters item: 14 with 28 elements: 1 10 23 39 57 77 99 122 146 171 194 214 234 254 273 293 313 335 360 384 407 430 451 470 487 501 511 514
%MSG
%MSG-i MTDParametersFromDD:  PreProcPath p1 07-May-2021 11:23:02 CEST  Run: 1 Event: 1
Adding PMTDParameters item: 15 with 28 elements: 0 0 0 0 0 0 0 0 0 2 6 7 8 8 7 6 5 0 0 0 0 0 0 0 0 0 0 0
%MSG
%MSG-i MTDParametersFromDD:  PreProcPath p1 07-May-2021 11:23:02 CEST  Run: 1 Event: 1
Adding PMTDParameters item: 16 with 28 elements: 0 0 0 0 0 0 0 0 0 0 5 6 7 8 8 7 6 2 0 0 0 0 0 0 0 0 0 0
%MSG
%MSG-i MTDParametersFromDD:  PreProcPath p1 07-May-2021 11:23:02 CEST  Run: 1 Event: 1
Adding PMTDParameters item: 17 with 28 elements: 0 0 0 0 0 0 0 0 0 0 4 6 7 8 7 7 6 3 0 0 0 0 0 0 0 0 0 0
%MSG
%MSG-i MTDParametersFromDD:  PreProcPath p1 07-May-2021 11:23:02 CEST  Run: 1 Event: 1
Adding PMTDParameters item: 18 with 28 elements: 0 0 0 0 0 0 0 0 0 3 6 7 7 8 7 6 4 0 0 0 0 0 0 0 0 0 0 0
%MSG
```